packages/approval-controller/src/ApprovalController.ts

```diff
@@ -427,9 +529,42 @@ export class ApprovalController extends BaseControllerV2<
    *
    * @param id - The id of the approval request.
    * @param value - The value to resolve the approval promise with.
+   * @param options - Options bag.
+   * @returns A promise that either resolves once a result is provided by
+   * the creator of the approval request, or immediately if `options.waitForResult`
+   * is `false` or `undefined`.
    */
-  accept(id: string, value?: unknown): void {
-    this._deleteApprovalAndGetCallbacks(id).resolve(value);
+  accept(
+    id: string,
+    value?: unknown,
+    options?: AcceptOptions,
+  ): Promise<AcceptResult> {
+    const approval = this.get(id) as ApprovalRequest<any>;
```

Comment:
I'd be explicit about why we are casting here.

If we move this line underneath `_deleteApprovalAndGetCallbacks` and add a comment saying that the previous method would have thrown if an approval doesn't exist for that id (although that's technically not true either, as here we are getting the `ApprovalRequest` from the state, but `_deleteApprovalAndGetCallbacks` gets the promise callbacks from a private field).

Also we can narrow it down to `as ApprovalRequest<ApprovalRequestData>`. In this case I'd prefer a `!`, but those are banned by the linter and that'd require an extra line so... `as` is fine.

Reply:
A workaround that is probably not worth it in my opinion is to have `_deleteApprovalAndGetCallbacks` changed (both in name and shape) to return both the callbacks and the `ApprovalRequest` that is deleting, and throw if either of them is not present.

That would save you from having to cast anything.

Reply:
It's actually not half bad once implemented, but I'll be happy with just a comment justifying the assertion.

```
accept(
    id: string,
    value?: unknown,
    options?: AcceptOptions,
  ): Promise<AcceptResult> {
const {
      approvalRequest,
      callbacks: { resolve: resolveRequestPromise },
    } = this._deleteApprovalAndGetRequestAndCallbacks(id);

...

private _deleteApprovalAndGetRequestAndCallbacks(id: string) {
    const approvalRequest = this.get(id);
    const callbacks = this._approvals.get(id);
    if (!approvalRequest || !callbacks) {
      throw new ApprovalRequestNotFoundError(id);
    }

    this._delete(id);
    return {
      callbacks,
      approvalRequest,
    };
  }
```

Reply:
I thought about doing the same, but I think it's overkill for my use case.

I can't move the cast as it's deleted so the `get` call after would return undefined.

Added a comment though, and used the better type for the template.